### DISCUSSION TITLE
Cyrillic support in paths and folders

### DISCUSSION THREAD

**Discussion by TarLink48** [AlexandreRouma/SDRPlusPlus#639] [Q&A]
Hello ! Please tell me if the program supports Cyrillic in paths and folder names or is it just my problem? Thank you ...
![Снимок](https://user-images.githubusercontent.com/99340373/153216781-2806c10f-b768-46ef-8488-f7a05411b02c.PNG)

**Comment by AlexandreRouma** [comment]
Hello,
Unicode is indeed not supported, please make sure to use only (extended) ascii for paths or strings.

**Reply by TarLink48** [reply]
> please make sure to use only (extended) ascii for paths or strings

Thanks for the answer! Unfortunately, I don't really understand what needs to be done, I communicate through a translator, renamed the folder from Russian to English and everything works, unfortunately this is very inconvenient in my case...

**Comment by TarLink48** [comment]
Unfortunately, the Frequency Manager has the same thing
![Снимок](https://user-images.githubusercontent.com/99340373/153234524-f870f52f-032f-4637-b745-d6e663d14a83.PNG)

**Reply by AlexandreRouma** [reply]
That's what I said. Only ascii and extended ascii is supported. Cyrilic is Unicode and is NOT supported.

**Reply by TarLink48** [reply]
> That's what I said. Only ascii and extended ascii is supported. Cyrilic is Unicode and is NOT supported.

It's a shame, thanks for the answer!

**Reply by sannysanoff** [reply]
> That's what I said. Only ascii and extended ascii is supported. Cyrilic is Unicode and is NOT supported.

Hi, @AlexandreRouma 

plz see this: https://github.com/sannysanoff/SDRPlusPlus/commit/55695e752727f88f664ec27d851270cf13571357
this enabled it, tested on linux + directory selection dialog + russian paths on disk.

best regards,
san.

**Reply by AlexandreRouma** [reply]
None of the strings in SDR++ expect UTF-8. You may be able to display some of it but it WILL create bugs in random places that do not expect multiple bytes per character.

**Reply by TarLink48** [reply]
> this enabled it, tested on linux + directory selection dialog + russian paths on disk.

Thanks ! If suddenly it appears for WIndows , please let me know , this is one of the reasons why I don't use SDR++

**Reply by sannysanoff** [reply]
> None of the strings in SDR++ expect UTF-8. You may be able to display some of it but it WILL create bugs in random places that do not expect multiple bytes per character.

but this fix does not do anything wth how strings are stored, manipulated, how they are imported or exported, or how strlen() works, because it's all already there. It just adds missing glyphs on render, which otherwise would render as "?". In other words, it will neither add any new bugs, nor fix existing ones. 

![image](https://user-images.githubusercontent.com/952071/153680527-1bfcf003-e471-4e85-aeee-e5e3bb5a8930.png)

Can't find other place besides mentioned where user can enter utf-8.

**Reply by TarLink48** [reply]
> Can't find other place besides mentioned where user can enter utf-8.


I don 't know if I got it right or not , that 's where I have a problem:

**Comment by TarLink48** [comment]
![Снимок](https://user-images.githubusercontent.com/99340373/153682746-c143d35e-37eb-4b25-a84a-08f2d0b59fb0.PNG)
![Снимок2](https://user-images.githubusercontent.com/99340373/153682748-9d3181c3-8e46-4ece-b647-b13430a7f61b.PNG)

**Reply by sannysanoff** [reply]
![image](https://user-images.githubusercontent.com/952071/153684793-13999645-c3af-404c-9456-aa812061f313.png)

![image](https://user-images.githubusercontent.com/952071/153684823-b5df6684-76c9-4c6a-b873-ca6f42174d21.png)

looks right. Screenshot made on windows.

**Reply by TarLink48** [reply]
> looks right. Screenshot made on windows.

It looks good , without skills , as I understand it , a person like me will not be able to get it on his own ?

**Reply by sannysanoff** [reply]
> will not be able to get it on his own

find me on telegram, we're getting close to offtopic.
